software/src/modules/ac011k/ac011k.cpp

```diff
@@ -1434,50 +1418,120 @@ void AC011K::loop()
                                 logger.printfln("   handshake complete, flash erase next");
                                 RemoteUpdate[7] = 2; // flash erase
                                 sendCommand(RemoteUpdate, sizeof(RemoteUpdate), sendSequenceNumber++);
-                            } else if(FlashVerify[7] == 4) { // flash verify
+                            } else if(FlashBuffer[7] == 4) { // flash verify
                                 logger.printfln("   handshake complete, flash verify next");
                                 ready_for_next_chunk = true;
                             }
                             break;
                         case 2: // flash erase
                             logger.printfln("   flash erase complete, flash write next");
                             ready_for_next_chunk = true;
+                            if (LittleFS.exists(GD_FIRMWARE_FILE_NAME)) {
+                                task_scheduler.scheduleOnce([this](){
+                                    logger.printfln("Start flashing the GD chip with Firmware from LittleFS.");
+                                    GD_firmware_file = LittleFS.open(GD_FIRMWARE_FILE_NAME);
+                                    if (GD_firmware_file.available()) {
+                                        GD_firmware_filesize = GD_firmware_file.size() - GD_FIRMWARE_SKIP;
+                                        GD_firmware_flash_interval = GD_firmware_filesize / 10;
+                                        GD_firmware_nextPercent = GD_firmware_flash_interval;
+                                        GD_firmware_file.seek(GD_FIRMWARE_SKIP); // the actual to be flashed firmware starts at offset 0x8000
+                                        gd_flash_index = GD_FIRMWARE_SKIP;
+                                        GD_firmware_file_bytes_read = GD_firmware_file.read(((uint8_t *)FlashBuffer)+11, 800); // 800 bytes is the max chunk size that the GD can flash
+                                        flashChunk(FlashBuffer, gd_flash_index, GD_firmware_file_bytes_read);
+                                    }
+                                }, 5000);
+                            } else {
+                                if (GD_firmware_len > 0) {
+                                    task_scheduler.scheduleOnce([this](){
+                                        logger.printfln("Start flashing the GD chip with build in firmware.");
+                                        GD_firmware_filesize = GD_firmware_len - GD_FIRMWARE_SKIP;
+                                        GD_firmware_flash_interval = GD_firmware_filesize / 10;
+                                        GD_firmware_nextPercent = GD_firmware_flash_interval;
+                                        gd_flash_index = GD_FIRMWARE_SKIP;
+                                        memcpy(((uint8_t *)FlashBuffer)+11, GD_firmware + gd_flash_index, 800); // 800 bytes is the max chunk size that the GD can flash
```

Comment:
This causes in combination with the ifdef else in line 38 a warning that I've no idea how to get rid off:
It is not really an issue, because this code path will not be reached in case the firmware is not embedded.

```
In lambda function,
    inlined from 'static void std::_Function_handler<void(_ArgTypes ...), _Functor>::_M_invoke(const std::_Any_data&, _ArgTypes&& ...) [with _Functor = AC011K::loop()::<lambda()>; _ArgTypes = {}]' at ~/.platformio/packages/toolchain-xtensa-esp32/xtensa-esp32-elf/include/c++/8.4.0/bits/std_function.h:297:37:
src/modules/ac011k/ac011k.cpp:1472:47: warning: 'void* memcpy(void*, const void*, size_t)' reading 800 bytes from a region of size 0 [-Wstringop-overflow=]
                                         memcpy(((uint8_t *)FlashBuffer)+11, GD_firmware + gd_flash_index, 800); // 800 bytes is the max chunk size that the GD can flash
                                         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Why not also remove this line with an ifdef?